eidas_proxy_service/datamodels.py

```diff
@@ -0,0 +1,129 @@
+"""
+Lightweight data models.
+
+The declaration of a data model is as simple as:
+
+.. code:: python
+
+    class User(DataModel):
+        name = None  # type: str
+        age = None  # type: int
+
+        def validate(self) -> None:
+            ...
+
+Uppercase attributes or attributes starting with an underscore are not considered as fields.
+"""
+import types
+from abc import ABCMeta, abstractmethod
+from collections import OrderedDict
+from typing import Any, Dict, Iterator, List, Mapping, Tuple, Type
+
+from django.core.exceptions import ValidationError
+
+__all__ = ['DataModel']
+
+
+class DataModelMeta(ABCMeta):
+    """Metaclass for data models."""
+
+    @classmethod
+    def __prepare__(mcs, __name: str, __bases: Tuple[type, ...], **kwargs: Any) -> Mapping[str, Any]:
+        # To preserve the order of class attributes. Not needed in Python 3.6+.
+        return OrderedDict()
+
+    def __new__(mcs, name: str, bases: Tuple[type, ...], namespace: Dict[str, Any]) -> Type:
+        if bases and DataModel not in bases:
```

Comment:
This is a reference to a different model, is this metaclass really needed?

Reply:
The alternative to the metaclass is to make the list of DataModel fields always explicit, which is not consistent with Django models and Python 3.7 dataclasses. Should I proceed with that?

Reply:
Metaclass is gone. All models list their fields explicitly.